tests/4.5/target_teams_distribute/test_target_teams_distribute_defaultmap.F90

```diff
@@ -0,0 +1,422 @@
+!===--- test_target_teams_distribute_defaultmap.F90-------------------------===//
+!
+! OpenMP API Version 4.5 Nov 2015
+!
+! This test uses the defaultmap clause on a target teams distribute directive.
+! This tests the following scalars: char, short, int, float, double, and enum.
+! Both using the clause defaultmap(tofrom:scalar) is used. When it is used,
+! the test tests the to nature by setting arrays to the value.  Then it is also
+! tested that, as opposed to the default action on scalars which is to first-
+! privatize them, they are shared and returned to the host.
+!
+! It also tests the default operation of treating scalars without the defaultmap
+! clause.  The test first tests the privatization of the firstprivatized
+! scalars and then separately tests the proper initialization of them separately
+!
+!===------------------------------------------------------------------------===//
+
+#include "ompvv.F90"
+
+#define N 1024
+
+PROGRAM test_target_teams_distribute_defaultmap
+  USE iso_fortran_env
+  USE ompvv_lib
+  USE omp_lib
+  implicit none
+  LOGICAL :: isSharedEnv
+  OMPVV_TEST_OFFLOADING
+  OMPVV_TEST_AND_SET_SHARED_ENVIRONMENT(isSharedEnv)
+  IF (isSharedEnv) THEN
+     OMPVV_WARNING("Shared memory environment. Scalars are not copied")
+     OMPVV_WARNING("but modified. Defaultmap off test is inconclusive")
+  END IF
+  OMPVV_TEST_VERBOSE(test_defaultmap_on() .ne. 0)
+  OMPVV_TEST_VERBOSE(test_defaultmap_off() .ne. 0)
+
+  OMPVV_REPORT_AND_RETURN()
+CONTAINS
+  INTEGER FUNCTION test_defaultmap_on()
+    INTEGER :: errors, x
+    CHARACTER :: scalar_char
+    BYTE :: scalar_byte
+    INTEGER(kind = 2) :: scalar_short
+    INTEGER(kind = 4) :: scalar_int
+    INTEGER(kind = 8) :: scalar_long_int
+    REAL(kind = 4) :: scalar_float
+    REAL(kind = 8) :: scalar_double
+    LOGICAL :: scalar_logical
+    LOGICAL(kind = 4) :: scalar_logical_kind4
+    LOGICAL(kind = 8) :: scalar_logical_kind8
+    COMPLEX :: scalar_complex
+    COMPLEX(kind = 16) :: scalar_double_complex
+
+    CHARACTER,DIMENSION(N) :: char_array
```

Comment:
Also, It was also discussed in Issue #41 that Characters are not considered scalar for Fortran: 
Section 1.2.6, page 11:
For Fortran: A scalar variable with intrinsic type, as defined by the base language,
excluding character type.